zulip_bots/zulip_bots/bots/trivia_quiz/test_trivia_quiz.py

```diff
@@ -69,3 +74,49 @@ def test_answers(self) -> None:
                    return_value=json.dumps(quiz)):
             with patch('zulip_bots.bots.trivia_quiz.trivia_quiz.start_new_quiz') as mock_new_quiz:
                 self._test('answer Q001 A', '**CORRECT!** Amphibian :tada:')
+
+    def test_handle_answer(self) -> None:
+        bot_handler = StubBotHandler()
+        quiz_payload = read_bot_fixture_data('trivia_quiz', 'test_new_question')['response']
+        with patch('random.shuffle'):
+            quiz = get_quiz_from_payload(quiz_payload)
+
+        # create test initial storage
+        bot_handler.storage.put('Q001', json.dumps(quiz))
+
+        # test for a correct answer
+        start_new_question, response = handle_answer(quiz, 'A', 'Q001', bot_handler)
+        self.assertTrue(start_new_question)
+        self.assertEqual(response, '**CORRECT!** Amphibian :tada:')
+
+        # test for an incorrect answer
+        start_new_question, response = handle_answer(quiz, 'D', 'Q001', bot_handler)
+        self.assertFalse(start_new_question)
+        self.assertEqual(response, '**WRONG!** D is not correct :disappointed:')
+
+        # create test storage for a question which has been incorrectly answered twice
+        quiz['answered_options'].append('C')
+        quiz['answered_options'].append('B')
+        bot_handler.storage.put('Q001', json.dumps(quiz))
```

Comment:
Now you've added `update_quiz`, maybe test that first, and then use it in this test method? Is that reasonable, @showell ?